test/shared/useTask.test.ts

```diff
@@ -0,0 +1,163 @@
+import test, { Test } from 'tape';
+import AbortController from 'abort-controller';
+import { useTask, Task } from '@/index';
+
+test('it can succeed with an asynchronous callback', async (t: Test) => {
+  // Given a "pending" async task that returns a number.
+  const task = useTask(async () => {
+    t.equal(task.getStatus(), 'running');
+    return 42;
+  });
+  t.equal(task.getResult(), undefined);
+  t.equal(task.getStatus(), 'pending');
+
+  // When we run the task.
+  await task.run();
+
+  // Then we get the right result and it was marked as successful.
+  t.equal(task.getResult(), 42);
+  t.equal(task.getStatus(), 'successful');
+  t.equal(task.getError(), undefined);
+});
+
+test('it can succeed with an synchronous callback', async (t: Test) => {
+  // Given a "pending" task that returns a number.
+  const task = useTask(() => {
+    t.equal(task.getStatus(), 'running');
+    return 42;
+  });
+  t.equal(task.getResult(), undefined);
+  t.equal(task.getStatus(), 'pending');
+
+  // When we run the task.
+  await task.run();
+
+  // Then we get the right result and it was marked as successful.
+  t.equal(task.getResult(), 42);
+  t.equal(task.getStatus(), 'successful');
+  t.equal(task.getError(), undefined);
+});
+
+test('it can fail', async (t: Test) => {
+  // Given a "pending" task that throws an error.
+  const task = useTask(() => {
+    t.equal(task.getStatus(), 'running');
+    throw new Error('Test Task Failure');
+  });
+  t.equal(task.getResult(), undefined);
+  t.equal(task.getStatus(), 'pending');
+
+  // When we run the task.
+  try {
+    await task.run();
+  } catch (error) {
+    // Then the task is marked as failed and we kept track of the error.
+    t.equal(task.getStatus(), 'failed');
+    t.equal(task.getResult(), undefined);
+    t.equal(task.getError(), error);
+    return;
+  }
+
+  t.fail('Test task should have failed');
+});
+
+test('it can be aborted using an AbortController', async (t: Test) => {
+  // Given a test task that returns a number after 100ms.
+  const task = useTask(async () => {
+    await new Promise((resolve) => setTimeout(resolve, 100));
+    return 42;
+  });
+
+  // And an abort controller used to cancel the task.
+  const abortController = new AbortController();
+
+  // When we run the task and abort after 10ms.
+  setTimeout(() => abortController.abort(), 10);
+  try {
+    await task.run({ signal: abortController.signal });
+  } catch (error) {
+    // Fail silently...
+  }
+
+  // Then the task was marked as canceled.
+  t.equal(task.getStatus(), 'canceled');
+  t.equal(task.getResult(), undefined);
+  t.equal((task.getError() as Event).type, 'abort');
+});
+
+test('it can be reset', async (t: Test) => {
+  // Given a test task that ran successfully.
```

Comment:
I'd convert all these comments to `t.comment('Given a task that runs successfully')` since then you see this as part of the test output and if it fails in the middle you know how far it got.

Reply:
I'd love to have a chat with you about that but I strongly disagree on that one.

I think comments provide the perfect visual separation in tests in a way that gives them breathing room and makes them more enjoyable to read.

Example: With comments.

<img width="775" alt="CleanShot 2022-04-19 at 08 58 27@2x" src="https://user-images.githubusercontent.com/3642397/163955052-4fdaa612-f9a6-4834-896e-ec455ed0a6ec.png">

Without comments. (Not to mention how bad it would look on multi-line comments).

<img width="868" alt="CleanShot 2022-04-19 at 08 59 09@2x" src="https://user-images.githubusercontent.com/3642397/163955066-49a0fc3c-8aec-4a65-a9b2-2dc3680177f4.png">

I'd be potentially willing to make the change if `t.comment` was actually providing value but I honestly find the logging of `tape` quite messy (even using something like `tap-spec`) and it just gets lost in a sea of console lines you don't care about when things are successful (that was also part of the reason I started that jest PR that we discarded).

In any case, this is something that would need to be done on a separate PR but I really think the purpose of these comments is not to be logged somewhere, but to **provide a better experience to the person reading the tests**, not reading the logs.